src/reducers/map.js

```diff
@@ -1,19 +1,33 @@
 import * as types from '../constants/actionTypes';
+import SelectorData from '../containers/MapExplorer-SelectorData.jsx';
 
-const initialState = { selectors: [] };
+const initialState = { selectors: [new SelectorData()] };
 
 export function map(state = initialState, action) {
   switch (action.type) {
     case types.ADD_MAP_SELECTOR:
       let newSelectors = state.selectors.slice();
-      newSelectors.push(Math.floor(Math.random() * 100));
-      return {
-        selectors: newSelectors
-      };
+      newSelectors.push(new SelectorData());
+      return Object.assign({}, state, {
+        selectors: newSelectors,
+      });
```

Comment:
@shaunanoordin You could use object spread operator here: 
`return {...state, selectors:  newSelectors,};`